cypress-tests/cypress/support/commands.js

```diff
@@ -888,26 +888,38 @@ Cypress.Commands.add(
         "api-key": globalState.get("apiKey"),
       },
       body: customerCreateBody,
+      failOnStatusCode: false,
     }).then((response) => {
-      globalState.set("customerId", response.body.customer_id);
-      logRequestId(response.headers["x-request-id"]);
+      if (response.status === 200) {
+        globalState.set("customerId", response.body.customer_id);
+        logRequestId(response.headers["x-request-id"]);
 
-      cy.wrap(response).then(() => {
-        expect(response.body.customer_id, "customer_id").to.not.be.empty;
-        expect(customerCreateBody.email, "email").to.equal(response.body.email);
-        expect(customerCreateBody.name, "name").to.equal(response.body.name);
-        expect(customerCreateBody.phone, "phone").to.equal(response.body.phone);
-        expect(customerCreateBody.metadata, "metadata").to.deep.equal(
-          response.body.metadata
-        );
-        expect(customerCreateBody.address, "address").to.deep.equal(
-          response.body.address
+        cy.wrap(response).then(() => {
+          expect(response.body.customer_id, "customer_id").to.not.be.empty;
+          expect(customerCreateBody.email, "email").to.equal(
+            response.body.email
+          );
+          expect(customerCreateBody.name, "name").to.equal(response.body.name);
+          expect(customerCreateBody.phone, "phone").to.equal(
+            response.body.phone
+          );
+          expect(customerCreateBody.metadata, "metadata").to.deep.equal(
+            response.body.metadata
+          );
+          expect(customerCreateBody.address, "address").to.deep.equal(
+            response.body.address
+          );
+          expect(
+            customerCreateBody.phone_country_code,
+            "phone_country_code"
+          ).to.equal(response.body.phone_country_code);
+        });
+      } else if (response.status === 400) {
+        expect(response.body.error.code).to.equal("IR_12");
+        expect(response.body.error.message).to.equal(
+          "Customer with the given `customer_id` already exists"
```

Comment:
there exist many other 4xx errors. this assertion will always be checked if status is 400. you need to validate before checking these assertions.

first verify if the error message is related to duplication. you can do something like:

```js
if (response.body.error.message).includes("already exists") {
	// check if it is the `customer_id` that exist and then make the assertions accordingly
	// also, remember to add names to that. `(...error.code, "error_code")`
}
```

Reply:
also, add an `else` condition at the end that actually throws an error if we get anything other than the mentioned we'll get to know about it